passiveData/src/commonMain/kotlin/org/sagebionetworks/assessmentmodel/passivedata/asyncaction/AsyncActionConfiguration.kt

```diff
@@ -0,0 +1,31 @@
+package org.sagebionetworks.assessmentmodel.passivedata.asyncaction
+
+/**
+ *  `AsyncActionConfiguration` defines general configuration for an asynchronous background action
+ * that should be run in the background. Depending upon the parameters and how the action is set
+ * up, this could be something that is run continuously or else is paused or reset based on a
+ * timeout interval.
+ *
+ * The configuration is intended to be a serializable object and does not call services, record
+ * data, or anything else.
+ *
+ * - seealso: `AsyncActionController`.
+ *
+ */
+
+
+interface AsyncActionConfiguration {
+    /// A short string that uniquely identifies the asynchronous action within the task.
+    val identifier: String
+
+    /// The type of the async action.
+    val typeName: String
```

Comment:
This is here on the Swift side b/c Swift `Codable` isn't polymorphic so I support polymorphism using the `JsonModel` library. For that library to work, the Codable objects have to have a key that gets encoded as "type" **and** the deserialization factory decodes them by using this key in a registered mapping to find the `Type` of the object. 

Since `kotlinx.serialization` **is** polymorphic, but the object classes do not have access to their "type" key (in the JSON model) this isn't necessary, is it?

ie. We don't use `typeName` in the serialization of AssessmentModelKMM and instead rely upon polymorphic serialization. We **do** use `typeName` in SageResearch b/c that uses the `JsonModel.SerializationFactory` to support polymorphism.